docs/essentials/exceptions/exceptions-best-practices.md

```diff
@@ -192,29 +97,15 @@ The following list contains guidelines for creating your own exceptions and when
-  Public Class MyFileNotFoundException
-      Inherits Exception
-  End Class
-  ```
-
 - In C# and C++, use at least the three common constructors when creating your own exception classes: the default constructor, a constructor that takes a string message, and a constructor that takes a string message and an inner exception. For an example, see How to: Create User-Defined Exceptions.
 
-  - [Exception()](https://msdn.microsoft.com/library/9f9b3wha), which uses default values.
+  - [Exception()](http://dotnet.github.io/api/System.Exception.html#System_Exception__ctor), which uses default values.
 
-  - [Exception(String)](https://msdn.microsoft.com/library/48ca3hhw), which accepts a string message.
+  - [Exception(String)](http://dotnet.github.io/api/System.Exception.html#System_Exception__ctor_System_String_), which accepts a string message.
 
-  - [Exception(String, Exception)](https://msdn.microsoft.com/library/804f22sf), which accepts a string message and an inner exception.
+  - [Exception(String, Exception)](http://dotnet.github.io/api/System.Exception.html#System_Exception__ctor_System_String_System_Exception_), which accepts a string message and an inner exception.
 
-- When you create user-defined exceptions, you must ensure that the metadata for the exceptions is available to code that is executing remotely, including when exceptions occur across app domains. For example, suppose App Domain A creates App Domain B, which executes code that throws an exception. For App Domain A to properly catch and handle the exception, it must be able to find the assembly that contains the exception thrown by App Domain B. If App Domain B throws an exception that is contained in an assembly under its application base, but not under App Domain A's application base, App Domain A will not be able to find the exception, and the common language runtime will throw a [FileNotFoundException](https://msdn.microsoft.com/library/system.io.filenotfoundexception) exception. To avoid this situation, you can deploy the assembly that contains the exception information in two ways:
+- When you create user-defined exceptions, you must ensure that the metadata for the exceptions is available to code that is executing remotely, including when exceptions occur across app domains. For example, suppose App Domain A creates App Domain B, which executes code that throws an exception. For App Domain A to properly catch and handle the exception, it must be able to find the assembly that contains the exception thrown by App Domain B. If App Domain B throws an exception that is contained in an assembly under its application base, but not under App Domain A's application base, App Domain A will not be able to find the exception, and the common language runtime will throw a [FileNotFoundException](http://dotnet.github.io/api/System.IO.FileNotFoundException.html) exception. To avoid this situation, you can deploy the assembly that contains the exception information in two ways:
```

Comment:
We could keep the AppDomain content, but frame it as "On .NET runtimes that implement Application Domains ..." I'd be happy with that.